ui/src/features/Canvas/index.tsx

```diff
@@ -182,6 +188,13 @@ const Canvas: React.FC<Props> = ({
           onClose={handleCloseContextmenu}
         />
       )}
+      {showBeforeDeleteDialog && (
+        <NodeDeletionDialog
+          showBeforeDeleteDialog={showBeforeDeleteDialog}
+          deferredDeleteRef={deferredDeleteRef}
+          onDialogClose={handleDeleteDialogClose}
+        />
+      )}
```

Comment:
I just realized this shouldn't be inside ReactFlow, and thus not be inside Canvas/index.
This should be in Editor/index 🙏🏻